python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+        return as many magnitudes as were loaded with the loadBandpassesFromFiles methods; it will
+        just return nonsense for magnitudes you did not actually ask for)
+
+        @param [out] magList is a list of magnitudes in the bandpasses stored in self.bandpassDict
+        """
+
+        if sedobj.wavelen is not None:
+
+            # If the Sed's wavelength grid agrees with self._wavelen_match to one part in
+            # 10^6, just use the Sed as-is.  Otherwise, copy it and resample it onto
+            # self._wavelen_match
+            if len(sedobj.wavelen)!=len(self._wavelen_match) or \
+            not numpy.allclose(sedobj.wavelen, self._wavelen_match, atol=0.0, rtol=1.0e-6):
+                dummySed = Sed(wavelen=sedobj.wavelen, flambda=sedobj.flambda)
+                dummySed.resampleSED(wavelen_match=self._bandpassDict.values()[0].wavelen)
+            else:
+                dummySed = sedobj
+
+
+            #for some reason, moving this call to flambdaTofnu()
+            #to a point earlier in the
+            #process results in some SEDs having 'None' for fnu.
+            #
+            #I looked more carefully at the documentation in Sed.py
+            #Any time you update flambda in any way, fnu gets set to 'None'
+            #This is to prevent the two arrays from getting out synch
+            #(e.g. renormalizing flambda but forgettint to renormalize fnu)
+            #
+            dummySed.flambdaTofnu()
+
+            if indices is not None:
+                magList = [numpy.NaN]*self._nBandpasses
+
+                magArray = dummySed.manyMagCalc(self._phiArray, self._wavelenStep, observedBandpassInd=indices)
+                for i,ix in enumerate(indices):
+                    magList[ix] = magArray[i]
+            else:
+                magList = dummySed.manyMagCalc(self._phiArray, self._wavelenStep)
+
+            return numpy.array(magList)
+
+        else:
+            return numpy.array([numpy.NaN]*self._nBandpasses)
+
+
+
+    def calcMagListFromSedList(self, sedList, indices=None):
+        """
+        Return a 2-D array of magnitudes from a SedList.
+        Each row will correspond to a different Sed, each column
+        will correspond to a different bandpass.
+
+        @param [in] sedList is a SedList containing the Seds
+        whose magnitudes are desired.
+
+        @param [in] indices is an optional list of indices indicating which bandpasses to actually
+        calculate magnitudes for.  Other magnitudes will be listed as 'None' (i.e. this method will
+        return as many magnitudes as were loaded with the loadBandpassesFromFiles methods; it will
+        just return nonsense for magnitudes you did not actually ask for)
+
+        @param [out] output_list is a 2-D numpy array containing the magnitudes
+        of each Sed (the rows) in each bandpass contained in this BandpassDict
+        (the columns)
+        """
+
```

Comment:
I can make the docstring more explicit.  I'll mention that, if the user matches their Sed's to the Bandpasses wavelength, that will be more efficient (that capability is explicitly included in the SedList class, where you can specify a wavelength_match grid on which to sample all of your SEDs).

You're right: I should change this method to use Sed._needResample here.